src/main/java/com/epam/test_generator/controllers/CaseController.java

```diff
@@ -114,16 +123,26 @@
     @ApiOperation(value = "Delete cases from suit", nickname = "removeCases")
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = "OK"),
-        @ApiResponse(code = 404, message = "Suit not found")
+        @ApiResponse(code = 404, message = "Suit not found"),
+        @ApiResponse(code = 400, message = "Suit doesn't contains some of cases")
     })
     @ApiImplicitParams({
-        @ApiImplicitParam(name = "suitId", value = "ID of suit which contains the case", required = true, dataType = "long", paramType = "path"),
-        @ApiImplicitParam(name = "suitDTO", value = "Suit object with casesIds", required = true, dataType = "SuitDTO", paramType = "body")
+        @ApiImplicitParam(name = "suitId", value = "ID of suit which contains the case",
+            required = true, dataType = "long", paramType = "path"),
+        @ApiImplicitParam(name = "removeCaseIds", value = "IDs of cases to be removed",
+            required = true, dataType = "Long[]", paramType = "body")
     })
     @RequestMapping(value = "/suits/{suitId}/cases", method = RequestMethod.DELETE, consumes = "application/json")
     public ResponseEntity<Void> removeCases(@PathVariable("suitId") long suitId,
-                                            @RequestBody @Valid SuitDTO suitDTO) {
+                                            @RequestBody Long[] removeCaseIds) {
+        SuitDTO suitDTO = suitService.getSuit(suitId);
+        if (!suitDTO.getCases().stream().map(CaseDTO::getId).collect(Collectors.toList())
+            .containsAll(
+                Arrays.asList(removeCaseIds))) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
         List<Long> caseIds = suitDTO.getCases().stream().map(CaseDTO::getId)
+            .filter(Arrays.asList(removeCaseIds)::contains)
```

Comment:
This check is redundant

Consider following code fragment:

        SuitDTO suitDTO = suitService.getSuit(suitId);
        List<Long> existentSuitCaseIds = suitDTO.getCases().stream()
            .map(CaseDTO::getId)
            .collect(Collectors.toList());
        List<Long> idsToRemove = Arrays.asList(removeCaseIds);

        if (!existentSuitCaseIds.containsAll(idsToRemove)) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }
        caseService.removeCases(suitId, idsToRemove);

        return new ResponseEntity<>(HttpStatus.OK);

Not only it avoids unnecessary stream iterations and ArrayList creations, but also makes **if** block self-explanatory.

Reply:
Changed method according to this comment